File: packages/OWASP/2025/top-10/improper-output-handling/ts/test_security.ts
import { LLM_1ntruder } from "./lib";
import { vulnerable_scenario } from "./vulnerable";
function assert_condition(condition: boolean, message: string): void { if (!condition) { throw new Error(message); } }
assert_condition(vulnerable_scenario("x <script").includes("<script"), "expected vulnerable marker");
assert_condition(!LLM_1ntruder.clean("Improper Output Handling", "x <script").includes("<script"), "expected marker removal");

File: packages/OWASP/2025/top-10/improper-output-handling/zig/test_security.zig
const std = @import("std"); const lib = @import("lib.zig"); const vulnerable = @import("vulnerable.zig");
test "vulnerability" { try std.testing.expect(std.mem.indexOf(u8, vulnerable.vulnerable_scenario("x <script"), "<script") != null); }
test "defense" { const cleaned = try lib.LLM_1ntruder.clean(std.testing.allocator, "Improper Output Handling", "x <script"); defer std.testing.allocator.free(cleaned); try std.testing.expect(std.mem.indexOf(u8, cleaned, "<script") == null); }
